### PR TITLE
feat: implement jasmine_test rule

### PR DESCRIPTION
**Pull request by thesayyn**

### CODE REVIEW

File: jasmine/defs.bzl [hunk 1]
```diff
@@ -0,0 +1,27 @@
+"Public API re-exports"
+
+
+load("//jasmine/private:jasmine_test.bzl", "lib")
+load("@aspect_rules_js//js:libs.bzl", "js_binary_lib")
+
+_jasmine_test = rule(
+    doc = """The Jasmine rule run tests under the Jasmine framework with Bazel.""",
```
Comment (jbedard) [inline jasmine/defs.bzl:8]: "Run tests under the Jasmine framework"?

File: jasmine/private/jasmine_test.bzl [hunk 2]
```diff
@@ -0,0 +1,62 @@
+load("@aspect_rules_js//js:libs.bzl", "js_binary_lib", "js_lib_helpers")
+load("@bazel_skylib//lib:dicts.bzl", "dicts")   
+
+_attrs = dicts.add(js_binary_lib.attrs, {
+    "entry_point": attr.label(
+        mandatory = True,
+    ),
+    "config": attr.label(
+        doc = "jasmine config file. See: ",
```
Comment (jbedard) [inline jasmine/private/jasmine_test.bzl:9]: see what?

File: jasmine/private/jasmine_test.bzl [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    files = js_lib_helpers.gather_files_from_js_providers(
+        targets = ctx.attr.data,
+        include_transitive_sources = ctx.attr.include_transitive_sources,
+        include_declarations = ctx.attr.include_declarations,
+        include_npm_linked_packages = ctx.attr.include_npm_linked_packages,
+    )
+    files.extend(ctx.files.data)
+
+    fixed_args = [
```
Comment (jbedard) [inline jasmine/private/jasmine_test.bzl:25]: I feel like this should be an argument with this value as the default. Do we have other `*_test` rules with something such as a `srcs` or `tests` argument?

Comment (thesayyn) [reply jasmine/private/jasmine_test.bzl:25]: I feel the same way.  Though https://github.com/bazelbuild/rules_nodejs/blob/8f2333cef700d2e94aead34cf710f07580896a68/packages/jasmine/jasmine_runner.js#L55 this was a constant in rules_nodejs and looks like nobody wanted to configure this.

Comment (jbedard) [reply jasmine/private/jasmine_test.bzl:25]: I guess it's easier to add it later then remove it later so stick with no argument for now?

Comment (thesayyn) [reply jasmine/private/jasmine_test.bzl:25]: Exactly.

**Comment by alexeagle** [comment]
can rebase on main now, I merged your initial one adding the toolchain

File: jasmine/defs.bzl (hunk 1 above)
Comment (thesayyn) [reply jasmine/defs.bzl]: fixed it.

File: jasmine/private/jasmine_test.bzl (hunk 2 above)
Comment (thesayyn) [reply jasmine/private/jasmine_test.bzl]: fixed it

File: e2e/simple/WORKSPACE [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Override http_archive for local testing
```
Comment (alexeagle) [inline e2e/simple/WORKSPACE:1]: our current setup in other repos has an examples/ folder which is not a nested workspace. I think we should stay consistent (if we're going to change, we should change all of them)

File: jasmine/BUILD.bazel [hunk 5]
```diff
@@ -28,4 +28,8 @@ bzl_library(
     name = "defs",
     srcs = ["defs.bzl"],
     visibility = ["//visibility:public"],
+    deps = [
+        "@aspect_rules_jasmine//jasmine/private:jasmine_test",
```
Comment (alexeagle) [inline jasmine/BUILD.bazel:32]: shouldn't need `@aspect_rules_jasmine` within the local repo

File: e2e/simple/WORKSPACE (hunk 4 above)
Comment (thesayyn) [reply e2e/simple/WORKSPACE:1]: Oh, I didn't mean this to be an example. I should rename it. I wanted to assert jasmine_test is usable from another workspace.

Comment (thesayyn) [reply e2e/simple/WORKSPACE:1]: I'll add an example folder as well.

File: jasmine/BUILD.bazel (hunk 5 above)
Comment (thesayyn) [reply jasmine/BUILD.bazel]: somehow stardoc gets confused and errors with 

```
Stardoc documentation generation failed: File jasmine/defs.bzl imported '@aspect_rules_jasmine//jasmine/private:jasmine_test.bzl', yet jasmine/private/jasmine_test.bzl was not found, even at roots [., external/aspect_rules_jasmine].
```

i am not sure what causes this.

Comment (thesayyn) [reply jasmine/BUILD.bazel]: Oh nvm. found the reason.

File: jasmine/repositories.bzl [hunk 6]
```diff
@@ -14,35 +14,30 @@ _ATTRS = {
...
-    name = "{name}",
-    data = [
-        ":node_modules/jasmine",
-        ":node_modules/jasmine-core"
-    ],
-    entry_point = ":runner.js",
+directory_path(
+    name = "jasmine_entrypoint",
+    directory = ":node_modules/jasmine/dir",
+    path = "bin/jasmine.js",
```
Comment (alexeagle) [inline jasmine/repositories.bzl:28]: did we lose all the sharding and test xml features from https://github.com/bazelbuild/rules_nodejs/blob/stable/packages/jasmine/jasmine_runner.js ?

Comment (thesayyn) [reply jasmine/repositories.bzl:28]: nope. being implemented over #7

Comment (alexeagle) [reply jasmine/repositories.bzl:28]: okay sounds good, one step at a time

**Review by alexeagle — APPROVED** [review]

**Pull request merged**